packages/as-proto-gen/src/generate/header.ts

```diff
@@ -0,0 +1,30 @@
+import { Version } from "google-protobuf/google/protobuf/compiler/plugin_pb";
+import { version as packageVersion } from "../../package.json";
+
+export function generateHeaderComment(compilerVersion: Version | undefined) {
+  let comment = "";
+
+  comment += `// Code generated by protoc-gen-as. DO NOT EDIT.\n`;
+  comment += `// Versions:\n`;
+  comment += `//   protoc-gen-as v${packageVersion}\n`;
+
+  if (compilerVersion) {
+    comment += `//   protoc        v${versionString(compilerVersion)}\n`;
+  }
+
+  return comment;
+}
+
+function versionString(version: Version | undefined): string {
```

Comment:
nit: could you write a small description of what this function does?

Reply:
Apart from adding a description, I've refactored the function so that it's more readable.